src/sema/symtable.rs

```diff
@@ -78,12 +78,12 @@ pub enum VariableUsage {
 }
 
 #[derive(Debug, Clone)]
-struct VarScope(HashMap<String, usize>, Option<HashSet<usize>>);
+pub struct VarScope(pub HashMap<String, usize>, Option<HashSet<usize>>);
 
 #[derive(Default, Debug, Clone)]
 pub struct Symtable {
     pub vars: IndexMap<usize, Variable>,
-    scopes: Vec<VarScope>,
+    pub scopes: Vec<VarScope>,
```

Comment:
Surely this vector is empty after resolving a function, all the scopes have been popped off again.

Reply:
The outermost scope of a function (which includes the arguments and named return values along with the local variables declared in the outermost scope) is not popped. 
So, currently I am using the scope that's left behind in the `scopes` field in `languageserver` module. 
Do you think there is a better place where I can do this? 

@seanyoung

Reply:
If the last scope is not popped then that's a bug. Also, that is not enough information, you need all the scopes, right?

Reply:
> If the last scope is not popped then that's a bug.
I just printed the `Namespace` struct from within the `languageserver` module. I can see that there is always 1 `VarScope` element in the `scopes` field. 

> you need all the scopes, right?
Yes, I need information about all the different lexical scopes of a function and the variables declared within each of the scopes.

Reply:
A new field with all the scopes might be better.

Reply:
I have made some changes. Now I have introduced a new field `popped_fields` (looking for a better name). The scopes popped by calling `leave_scope` method are now added to the `popped_scopes` - https://github.com/hyperledger/solang/pull/1560/files#diff-88601e831b596c4462675b562c54089bbcf53d14b625c8fb8a955eed6d6bb08aR194

All the lexical scopes but the outermost scope of a function are added to `popped_scopes` by the time `languageserver` module can use `Namespace` struct. 

I make use of these `popped_scopes` along with the outermost scope that has been left behind in the `languageserver` module. - https://github.com/hyperledger/solang/pull/1560/files#diff-bf2200d9cb4eee660f8ff53c8416979a1f0ebe5d5856609e7d7f46fe0c8e9846R1619-R1652

Reply:
#1560 has been merged, please see if you can the `active_scopes` into `ExprContext`. :pray: